### PR TITLE
linked url with goal modal

### LINKED ISSUE #158: Link goal modal with goal link
URL - `BASE_URL/goal/:goalId`
The above url should open the goal modal

### PR DESCRIPTION
**Pull request by pallabez**
### Issue:
Fixes #158
### Dev Tested:

- [x] Yes
- [ ] No

### Put the Feature behind Feature Flag

- [ ] Yes
- [x] No

### Follow-up Issues (if any):

> NOTE: After your PR is merged in main please verify its working on prod and raise a dev to main PR

### CODE REVIEW

File: components/GoalListTabEdit.vue [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <template>
   <tr v-if="goal">
-    <td @click="openCreateGoalModal(state.id)">{{ state.title }}</td>
+    <nuxt-link :to="`/goal/${props.goalId}`">
```
Comment (bhtibrewal) [inline components/GoalListTabEdit.vue:3]: while navigating here to the modal, can we pass it a reference to the current url we are at, so on close it sends us back to it

File: app/router.options.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import type { RouterConfig } from '@nuxt/schema'
+
+export default {
```
Comment (bhtibrewal) [inline app/router.options.ts:3]: this is what I was looking for few days back, how to do

Comment (pallabez) [reply app/router.options.ts:3]: Yeah. It was quite hidden in docs.

File: components/GoalListTabEdit.vue (hunk 1 above)
Comment (pallabez) [reply components/GoalListTabEdit.vue:3]: We can use router.back() from useRouter composable for that. Need to have a discussion regarding this tho.

Reference - https://nuxt.com/docs/api/composables/use-router

**Review by bhtibrewal — APPROVED** [review]
LGTM

**Pull request merged**